pkg/injection/namespace/bootstrapperconfig/bootstrapperconfig.go

```diff
@@ -0,0 +1,227 @@
+			return nil, errors.WithStack(err)
+		}
+
+		if _, err := endpointPropertiesBuilder.WriteString(fmt.Sprintf("%s=%s\n", dtingestendpoint.MetricsTokenSecretField, fields[dtingestendpoint.MetricsTokenSecretField])); err != nil {
+			return nil, errors.WithStack(err)
+		}
+	}
+
+	data := map[string][]byte{
+		endpoint.InputFileName: bytes.NewBufferString(endpointPropertiesBuilder.String()).Bytes(),
+	}
+
+	return data, nil
+}
+
+func (s *SecretGenerator) prepareFieldsForEndpoints(ctx context.Context, dk *dynakube.DynaKube) (map[string]string, error) {
+	fields := make(map[string]string)
+
+	tokens, err := k8ssecret.Query(s.client, s.apiReader, log).Get(ctx, client.ObjectKey{Name: dk.Tokens(), Namespace: s.namespace})
+	if err != nil {
+		return nil, errors.WithMessage(err, "failed to query tokens")
+	}
+
+	if dk.MetadataEnrichmentEnabled() {
+		if token, ok := tokens.Data[dtclient.DataIngestToken]; ok {
+			fields[dtingestendpoint.MetricsTokenSecretField] = string(token)
+		} else {
+			log.Info("data ingest token not found in secret", "dk", dk.Name)
+		}
+
+		if ingestUrl, err := dtingestendpoint.IngestUrlFor(dk); err != nil {
+			return nil, err
+		} else {
+			fields[dtingestendpoint.MetricsUrlSecretField] = ingestUrl
+		}
+	}
+
+	return fields, nil
+}
+
+func (s *SecretGenerator) preparePMC(ctx context.Context, dk dynakube.DynaKube) ([]byte, error) {
+	pmc, err := s.dtClient.GetProcessModuleConfig(ctx, 0)
+	if err != nil {
+		conditions.SetDynatraceApiError(dk.Conditions(), processmoduleconfigsecret.PMCConditionType, err)
+
+		return nil, err
+	}
+
+	tenantToken, err := k8ssecret.GetDataFromSecretName(ctx, s.apiReader, types.NamespacedName{
+		Name:      dk.OneAgent().GetTenantSecret(),
+		Namespace: dk.Namespace,
+	}, connectioninfo.TenantTokenKey, log)
+	if err != nil {
+		conditions.SetKubeApiError(dk.Conditions(), processmoduleconfigsecret.PMCConditionType, err)
+
+		return nil, err
+	}
+
+	pmc = pmc.
+		AddHostGroup(dk.OneAgent().GetHostGroup()).
+		AddConnectionInfo(dk.Status.OneAgent.ConnectionInfoStatus, tenantToken).
+		// set proxy explicitly empty, so old proxy settings get deleted where necessary
+		AddProxy("")
+
+	if dk.NeedsOneAgentProxy() {
+		proxy, err := dk.Proxy(ctx, s.apiReader)
+		if err != nil {
+			conditions.SetKubeApiError(dk.Conditions(), processmoduleconfigsecret.PMCConditionType, err)
+
+			return nil, err
+		}
+
+		pmc.AddProxy(proxy)
+
+		multiCap := capability.NewMultiCapability(&dk)
+		dnsEntry := capability.BuildDNSEntryPointWithoutEnvVars(dk.Name, dk.Namespace, multiCap)
+
+		if dk.FeatureNoProxy() != "" {
+			dnsEntry += "," + dk.FeatureNoProxy()
+		}
+
+		pmc.AddNoProxy(dnsEntry)
+	}
+
+	marshaled, err := json.Marshal(pmc)
+	if err != nil {
+		log.Info("could not marshal process module config")
+
+		return nil, err
+	}
+
+	return marshaled, err
+}
```

Comment:
nitpick: I could see this in a separate file, example: `pmc.go`, just to give it more structure

Reply:
done https://github.com/Dynatrace/dynatrace-operator/pull/4515/commits/726a2d29bf174578887380d34b448befd4f07779